src/app/icd/client/CheckInHandler.cpp

```diff
@@ -109,8 +109,13 @@ CHIP_ERROR CheckInHandler::OnMessageReceived(Messaging::ExchangeContext * ec, co
 
     if (refreshKey)
     {
-        // TODO: A new CASE session should be established to re-register the client using a new key. The registration will happen in
-        // CASE session callback
+        ICDRefreshKeyInfo * refreshKeyInfo = mpCheckInDelegate->OnKeyRefreshNeeded(clientInfo, mpICDClientStorage);
+        if (refreshKeyInfo == nullptr)
```

Comment:
Nothing in the API said returning null is OK.  This should be clearly documented, along with what null means (no more key refreshes possible right now due to resource constraints).

Reply:
I'll document that.